### PR TITLE
Add widget and visualization components

### PR DESCRIPTION
**Pull request by rihorn2**
Add the widget and generated javascript into the python distribution. Also add the source typescript into the repo in a separate directory. The widget implemented here maps from the InterpretML explanation.data output to the appropriate dashboard component data

### CODE REVIEW

**Comment by rihorn2** [comment]
@Harsha-Nori @interpret-ml  We can go over the details of this PR and the appropriate surface area for the widget as needed.

File: python/interpret/widget/ExplanationDashboard.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        self._mli_explanations = explanationObject.data(-1)["mli"]
+        local_explanation = self._find_first_explanation(ExplanationDashboardInterface.MLI_LOCAL_EXPLANATION_KEY)
+        global_explanation = self._find_first_explanation(ExplanationDashboardInterface.MLI_GLOBAL_EXPLANATION_KEY)
+        ebm_explanation = self._find_first_explanation(ExplanationDashboardInterface.MLI_EBM_GLOBAL_EXPLANATION_KEY)
+        dataset_explanation = self._find_first_explanation(ExplanationDashboardInterface.EXPLANATION_DATASET_KEY)
+
+        dataset_x = None
+        dataset_y = None
+        predicted_y = None
+        if dataset_explanation is not None:
```
Comment (rihorn2) [inline python/interpret/widget/ExplanationDashboard.py]: TODO: Ignore dataset explanation until it is reliably there and in usable format

File: python/interpret/widget/js/.npmrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+@vienna:registry=https://msdata.pkgs.visualstudio.com/_packaging/vienna-shared-ux/npm/registry/
```
Comment (rihorn2) [inline python/interpret/widget/js/.npmrc]: remove: this is no longer applicable in this repo

**Comment by wamartin-aml** [comment]
@rihorn2 can you point out the main files that should be reviewed? The +37800 lines is scary even knowing that a lot of it comes from generated javascript.

**Comment by interpret-ml** [comment]
Closing upon request from @rihorn2

**Pull request closed without merging**